raiden/raiden_service.py

```diff
@@ -573,13 +583,14 @@ def _register_payment_status(
             payment_type: PaymentType,
             balance_proof: BalanceProofUnsignedState,
     ):
-        self.targets_to_identifiers_to_statuses[target][identifier] = PaymentStatus(
-            payment_type=payment_type,
-            payment_identifier=identifier,
-            amount=balance_proof.transferred_amount,
-            token_network_identifier=balance_proof.token_network_identifier,
-            payment_done=AsyncResult(),
-        )
+        with self.payment_identifier_lock:
```

Comment:
This lock is not necessary, it's likewise around an atomic operation.

Reply:
As I understood, the lock was added to protect:

`this = self.targets_to_identifiers_to_statuses[target]`

and not

`this = self.targets_to_identifiers_to_statuses`

is that correct? have you considered making the lock more coarse to include all accesses to `targets_to_identifiers_to_statuses`?

Reply:
Hmm I don't understand the question. Yes the lock is to protect `self.targets_to_identifiers_to_statuses[target]`

Reply:
What is the lock protecting? Accesses to the value `targets_to_identifiers_to_statuses` itself or accesses to the values inside this dictionary?

IOW, if we get the expression:

```python
 payment_status = self.targets_to_identifiers_to_statuses[target].get(identifier)
```

and split it up to:

```python
statuses = self.targets_to_identifiers_to_statuses[target]
payment_status = statuses.get(identifier) 
```

where should the lock be placed?

```python
with self.payment_identifier_lock:
    statuses = self.targets_to_identifiers_to_statuses[target]
    payment_status = statuses.get(identifier) 
```
or:
```python

statuses = self.targets_to_identifiers_to_statuses[target]
with self.payment_identifier_lock:
    payment_status = statuses.get(identifier) 
```

?

Reply:
I see. At the time of writing this was only conceived to protect the same identifier for same target case. So the second option.

Reply:
have you considered making the lock more coarse to include all accesses to targets_to_identifiers_to_statuses?

Reply:
I see. Yes I agree due to the existence of the implicit write.